crates/e2e/src/builders.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use ink::codegen::TraitCallBuilder;
+use ink_env::{
+    call::{
+        utils::{
+            ReturnType,
+            Set,
+            Unset,
+        },
+        Call,
+        CallBuilder,
+        CreateBuilder,
+        ExecutionInput,
+        FromAccountId,
+    },
+    Environment,
+};
+use scale::Encode;
+
+/// The type returned from `ContractRef` constructors, partially initialized with the execution
+/// input arguments.
+type CreateBuilderPartial<E, Args, R> = CreateBuilder<
+    E,
+    Unset<<E as Environment>::Hash>,
+    Unset<u64>,
+    Unset<<E as Environment>::Balance>,
+    Set<ExecutionInput<Args>>,
+    Unset<ink_env::call::state::Salt>,
+    R,
+>;
+
+/// Shim onto the `CreateBuilder` to allow access to the `ExecutionInput` arguments.
+pub struct ConstructorBuilder<E: Environment, Args: Encode, R> {
+    inner: CreateBuilderPartial<E, Args, R>,
+}
```

Comment:
I don't really understand the need for this. You're exposing the `exec_input` getter anyways, and you should be able to use the `CreateBuilder` directly

Reply:
Originally I wanted to add the ability to *optionally* set the other properties e.g. `endowment`, `salt` etc but decided it was beyond the scope of this PR. However I kept the builder wrapper, which as you say is no longer necessary. 

I have changed it to a plain function and we can always add the builder back in the future if necessary.